controllers/certificaterequest/utils.go

```diff
@@ -20,17 +20,22 @@ import (
 	"context"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // SecretExists returns a boolean to the caller basd on the secretName and namespace args.
-func SecretExists(kubeClient client.Client, secretName, namespace string) bool {
-
+func SecretExists(kubeClient client.Client, secretName, namespace string) (bool, error) {
 	s := &corev1.Secret{}
-
 	err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, s)
-	return (err == nil)
+	if err == nil {
+		return true, nil
+	}
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	return false, err
```

Comment:
What is the point of this if statement? Why are we not returning the error here?

(I see the code difference, but maybe this warrants a comment in the code to explain the _why_ for our future selves)